buildSrc/src/main/kotlin/org/gradle/plugins/compile/AvailableJavaInstallations.kt

```diff
@@ -0,0 +1,62 @@
+package org.gradle.plugins.compile
+
+import org.gradle.api.JavaVersion
+import org.gradle.internal.jvm.Jvm
+import org.gradle.jvm.toolchain.internal.JavaInstallationProbe
+import org.gradle.jvm.toolchain.internal.LocalJavaInstallation
+import java.io.File
+
+class DefaultJavaInstallation(val current: Boolean) : LocalJavaInstallation {
+    private lateinit var name: String
+    private lateinit var javaVersion: JavaVersion
+    private lateinit var javaHome: File
+    private lateinit var displayName: String
+
+    override fun getName() = name
+    override fun getDisplayName() = displayName
+    override fun setDisplayName(displayName: String) { this.displayName = displayName }
+    override fun getJavaVersion() = javaVersion
+    override fun setJavaVersion(javaVersion: JavaVersion) { this.javaVersion = javaVersion }
+    override fun getJavaHome() = javaHome
+    override fun setJavaHome(javaHome: File) { this.javaHome = javaHome }
+    fun getToolsJar(): File? = Jvm.forHome(javaHome).toolsJar
```

Comment:
:o: This could be a `val`:

```kotlin
// either eager:
val toolsJar = Jvm.forHome(javaHome).toolsJar

// or computed on each call:
val toolsJar
    get() = Jvm.forHome(javaHome).toolsJar
```

Reply:
I ended up doing 
```
val toolsJar by lazy { Jvm.forHome(javaHome).toolsJar }
```